chain/network/src/routing.rs

```diff
@@ -327,17 +327,24 @@ impl RoutingTable {
             // nonce is greater than some threshold increase the lowest nonce to be at least
             // max nonce - threshold.
 
-            let (min_v, max_v) = routes.iter().fold((None, None), |(min_v, max_v), peer_id| {
-                let nonce = self.route_nonce.cache_get(&peer_id).cloned().unwrap_or(0usize);
-                let current = (nonce, peer_id.clone());
-                if min_v.is_none() || current < *min_v.as_ref().unwrap() {
-                    (Some(current), max_v)
-                } else if max_v.is_none() || *max_v.as_ref().unwrap() < current {
-                    (max_v, Some(current))
-                } else {
+            // Find node with min nonce and node with max nonce.
+            // If there are no routes both min_v and max_v will be None.
+            // If there is only one route max_v will be None.
+            let (min_v, max_v) =
+                routes.iter().fold((None, None), |(mut min_v, mut max_v), peer_id| {
+                    let nonce = self.route_nonce.cache_get(&peer_id).cloned().unwrap_or(0usize);
+                    let mut current = Some((nonce, peer_id.clone()));
+
+                    if min_v.is_none() || *current.as_ref().unwrap() < *min_v.as_ref().unwrap() {
```

Comment:
This is damn convoluted. Some comments describing basic scenarios (e.g. just one nonce, two nonces that increase, two nonces that decrease) and why those basic scenarios end up in the state we want would definitely help.

Reply:
Also, why `*current.as_ref().unwrap() < *min_v.as_ref().unwrap()` here, but just `max_v < current` below?

Reply:
I think it is just trying to compare the order of two things. The reason it looks complicated is because of `Option`.

Reply:
Yep, at first current could be None, but it ended being always Some and didn't fix that.

Reply:
Probably will implement a `min_max` function to get this logic out.